spec/forms/steps/income/income_payments_form_spec.rb

```diff
@@ -0,0 +1,231 @@
+              'interest_investment' => { 'amount_in_pounds' => '', 'frequency' => '' },
+              'student_loan_grant' => { 'amount_in_pounds' => '3.00', 'frequency' => 'annual' },
+              'board_from_family' => { 'amount_in_pounds' => '', 'frequency' => '' },
+              'rent' => { 'amount_in_pounds' => '2', 'frequency' => 'month' },
+              'financial_support_with_access' => { 'amount_in_pounds' => '', 'frequency' => '' },
+              'from_friends_relatives' => { 'amount_in_pounds' => '', 'frequency' => '' },
+              'other' => { 'amount_in_pounds' => '44', 'frequency' => 'week', 'details' => 'Side hustle' },
+            }
+          }
+        end
+
+        it 'is valid' do
+          expect(subject).to be_valid
+        end
+      end
+
+      context 'with invalid data' do
+        let(:form_data) do
+          {
+            'steps_income_income_payments_form' => {
+              'income_payments' => [''],
+              'types' => %w[maintenance student_loan_grant rent other],
+
+              'maintenance' =>  { 'amount_in_pounds' => '', 'frequency' => 'every week' },
+              'private_pension' => { 'amount_in_pounds' => '', 'frequency' => '' },
+              'state_pension' => { 'amount_in_pounds' => '', 'frequency' => '' },
+              'interest_investment' => { 'amount_in_pounds' => '', 'frequency' => '' },
+              'student_loan_grant' => { 'amount_in_pounds' => '3.00', 'frequency' => 'annual', 'details' => 'How?' },
+              'board_from_family' => { 'amount_in_pounds' => '', 'frequency' => '' },
+              'rent' => { 'amount_in_pounds' => '2', 'frequency' => 'month' },
+              'financial_support_with_access' => { 'amount_in_pounds' => '', 'frequency' => '' },
+              'from_friends_relatives' => { 'amount_in_pounds' => '', 'frequency' => '' },
+              'other' => { 'amount_in_pounds' => '44', 'frequency' => 'week', 'details' => 'Side hustle' },
+            }
+          }
+        end
+
+        it 'is invalid' do
+          expect(subject).not_to be_valid
+        end
+
+        it 'has error messages' do
+          expect(subject.errors.of_kind?('maintenance-amount_in_pounds', :greater_than)).to be(true)
+          expect(subject.errors.of_kind?('maintenance-frequency', :inclusion)).to be(true)
+          expect(subject.errors.of_kind?('student-loan-grant-details', :invalid)).to be(true)
+
+          # Error attributes should respond
+          expect(subject.send(:'maintenance-amount_in_pounds')).to eq '0.00'
+        end
+      end
+    end
+
+    context 'when `none` type' do
+      subject(:form) do
+        described_class.new(
+          crime_application: crime_application,
+          types: %w[none]
+        )
+      end
+
+      before do
+        subject.valid?
+      end
+
+      it 'saves nothing' do
+        expect(subject.crime_application.income_payments.size).to eq 0
+        expect(subject.save).to be true # Always true
```

Comment:
Good question - it's deliberate to show the form is a departure from other implementations in that the save will always occur, because by this point the actual records have already been inserted into `income_payments` and calling `subject.save` is part of the general form object lifecycle. Hope that makes sense. Do you think a fuller comment would be useful in the form code itself?